Mathlib/CategoryTheory/Sites/Sheafification.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Adjunction.Opposites
+import Mathlib.CategoryTheory.Adjunction.FullyFaithful
+import Mathlib.CategoryTheory.Sites.Sheaf
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+/-!
+
+# Sheafification
+
+Given a site `(C, J)` we define a typeclass `HasSheaf J A` saying that the inclusion functor from
+`A`-valued sheaves on `C` to presheaves admits a left adjoint (sheafification).
+-/
+
+universe v₁ v₂ u₁ u₂
+
+namespace CategoryTheory
+
+open Limits
+
+variable {C : Type u₁} [Category.{v₁} C] (J : GrothendieckTopology C)
+variable (A : Type u₂) [Category.{v₂} A]
+
+/--
+A proposition saying that the inclusion functor from sheaves to presheaves admits a left adjoint.
+-/
+abbrev HasWeakSheafify := Nonempty (IsRightAdjoint (sheafToPresheaf J A))
```

Comment:
This shouldn't be part of this PR, but in the future we should think about making `IsRightAdjoint` into a prop!